src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -947,11 +948,15 @@ zocl_xclbin_read_axlf(struct drm_zocl_dev *zdev, struct drm_zocl_axlf *axlf_obj,
 	zdev->zdev_xclbin->zx_refcnt = 0;
 	zocl_xclbin_set_uuid(zdev, &axlf_head.m_header.uuid);
 
+	vfree(aie_res);
```

Comment:
I dislike this duplicated code. Maybe we can use below code flow.

...
write_lock();
// Do everything that should be protected by the write lock

out0:
write_unlock();
if (!ret && kds_mode = 1) {
    // create cu sub-device
}
...

Reply:
Modified accordingly